akka-projection-grpc/src/test/java/akka/projection/grpc/replication/javdsl/ReplicationCompileTest.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (C) 2022 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.projection.grpc.replication.javdsl;
+
+import akka.Done;
+import akka.NotUsed;
+import akka.actor.typed.ActorSystem;
+import akka.grpc.GrpcClientSettings;
+import akka.grpc.javadsl.ServiceHandler;
+import akka.http.javadsl.Http;
+import akka.http.javadsl.ServerBinding;
+import akka.http.javadsl.model.HttpRequest;
+import akka.http.javadsl.model.HttpResponse;
+import akka.japi.function.Function;
+import akka.persistence.query.Offset;
+import akka.persistence.query.typed.EventEnvelope;
+import akka.persistence.typed.ReplicaId;
+import akka.persistence.typed.javadsl.ReplicatedEventSourcedBehavior;
+import akka.persistence.typed.javadsl.ReplicationContext;
+import akka.projection.ProjectionContext;
+import akka.projection.ProjectionId;
+import akka.projection.grpc.producer.EventProducerSettings;
+import akka.projection.grpc.replication.javadsl.Replica;
+import akka.projection.grpc.replication.javadsl.Replication;
+import akka.projection.grpc.replication.javadsl.ReplicationProjectionProvider;
+import akka.projection.grpc.replication.javadsl.ReplicationSettings;
+import akka.projection.javadsl.SourceProvider;
+import akka.projection.r2dbc.R2dbcProjectionSettings;
+import akka.projection.r2dbc.javadsl.R2dbcProjection;
+import akka.projection.javadsl.AtLeastOnceFlowProjection;
+import akka.stream.javadsl.FlowWithContext;
+
+import java.time.Duration;
+import java.util.HashSet;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.CompletionStage;
+
+public class ReplicationCompileTest {
+
+ interface MyCommand {}
+
+ static ReplicatedEventSourcedBehavior<MyCommand, Void, Void> create(ReplicationContext context) {
+   throw new UnsupportedOperationException("just a dummy factory method");
+ }
+
+  public static void start(ActorSystem<?> system) {
+   Set<Replica> otherReplicas = new HashSet<>();
+   otherReplicas.add(Replica.create(
```

Comment:
would it still work if adding the self Replica to otherReplicas? I'm thinking that it's convenient to be able to deploy the same code everywhere.

Reply:
I did that for config but currently it fails for programmatic as I thought that is easy enough to do your own filtering. Maybe it would be better with the convenience for programmatic bootstrap as well 👍

Reply:
Done c898d1e8c4c807582d9e18708cfa94032d5f31e4